app/controllers/CheckYourAnswersController.scala

```diff
@@ -53,11 +54,17 @@ class CheckYourAnswersController @Inject()(
   val logger: Logger = Logger(this.getClass())
 
   def onSubmit(nilReturn: Boolean): Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request =>
+     sessionRepository.set(request.userAnswers.copy(submitted = true))
```

Comment:
We are setting this flag to true before a successful submission of the returns which happens in ReturnsControllers onPageLoad function on line 62. So I think we should only set this if the submission has been successful.